account_payment_order/models/account_payment_order.py

```diff
@@ -461,39 +461,51 @@ def _prepare_move_line_partner_account(self, bank_line):
         return vals
 
     @api.multi
-    def generate_move(self):
+    def _create_reconcile_move(self, hashcode, blines):
+        self.ensure_one()
+        post_move = self.payment_mode_id.post_move
+        am_obj = self.env['account.move']
+        mvals = self._prepare_move(blines)
+        total_company_currency = total_payment_currency = 0
+        for bline in blines:
+            total_company_currency += bline.amount_company_currency
+            total_payment_currency += bline.amount_currency
+            partner_ml_vals = self._prepare_move_line_partner_account(
+                bline)
+            mvals['line_ids'].append((0, 0, partner_ml_vals))
+        trf_ml_vals = self._prepare_move_line_offsetting_account(
+            total_company_currency, total_payment_currency, blines)
+        mvals['line_ids'].append((0, 0, trf_ml_vals))
+        move = am_obj.create(mvals)
```

Comment:
You should add also a prepare method for this dictionary for proper hook design.

Reply:
I moved all the move dictionary preparation in the `_prepare_move` function.